lib/client/checks/api/checks-handler.ts

```diff
@@ -0,0 +1,54 @@
+import logger = require('../../../log');
+import { CheckResult } from '../types';
+import { Request, Response } from 'express';
+import { checksConfig } from '../index';
+
+export const handleChecksRoute = (
+  config: any,
+): ((_: Request, res: Response) => void) => {
+  return async (_: Request, res: Response) => {
+    try {
+      const { httpCheckService, preflightCheckStore } = await checksConfig(
+        config,
+      );
+      const checkResults: CheckResult[] = [];
+
+      const checks = await preflightCheckStore.getAll();
+      for (const check of checks) {
+        const checkResult = await httpCheckService.run(check.checkId);
+        checkResults.push(checkResult);
+      }
+      res.status(200).json(checkResults);
+    } catch (error) {
+      logger.error({ error }, 'Error executing http checks');
+      res.status(500).json({ error: error });
+    }
+  };
+};
+
+export const handleCheckIdsRoutes = (
```

Comment:
can't this handler be combined with the one above and you just check for the param existence to understand this is a specific checkId check vs general check?

Reply:
moved this sub-task to the next iteration.